### PR TITLE
refactor(dynamic-flows): remove QueryablePromise

### PR DESCRIPTION
**Pull request by JulianG-TW**
## 🖼 Context

**This is not necesary.**

But we can safely remove `QueryablePromise` and use a normal `Promise`.

Most persist-async related modules use a `QueryablePromise` instead of a native JS `Promise`.

But in `usePendingPromiseCounter` we are counting (persist-async) pending promises without using any of the additional methods (`isPending`, `isFulfilled`, `isRejected`)

## 🚀 Changes

Remove the `QueryablePromise` class/function and its use. Fix a test that incidentally used the `isPending` function.

## 🤔 Considerations

This would be a breaking change for consumers using JsonSchemaForm and expecting a QueryablePromise, but we couldn't find any usages within @transferwise org on GitHub.

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/jsonSchemaForm/persistAsyncSchema/basic/PersistAsyncBasicSchema.spec.js [hunk 1]
```diff
@@ -155,10 +155,10 @@ describe('Given a component for rendering persist async schemas', () => {
             enterValueAndBlur('200--ok--fast-5ms');
           });
 
-          it('should trigger onPersistAsync correctly', () => {
+          it('should trigger onPersistAsync correctly', async () => {
             expect(onPersistAsync).toHaveBeenCalledTimes(1);
             const firstArgumentOfFirstCall = onPersistAsync.mock.calls[0][0];
-            expect(firstArgumentOfFirstCall.isPending()).toBe(true);
+            await expect(firstArgumentOfFirstCall).resolves.not.toThrow();
```
Comment (JulianG-TW) [inline packages/dynamic-flows/src/jsonSchemaForm/persistAsyncSchema/basic/PersistAsyncBasicSchema.spec.js:161]: Technically, we were using `isPending` but only in a test.

**Comment by fergusjordantw** [comment]
> This is however a breaking change, for consumers using JsonSchemaForm and expecting a QueryablePromise.

Can we use codesearch to determine if any consumers are expecting to query the returned promise?

**Comment by JulianG-TW** [comment]
> Can we use codesearch to determine if any consumers are expecting to query the returned promise?

I just searched and I could not find any usages of either `isPending`, `isFulfilled` nor `isRejected`.
The function names are very common, so they DO show up in codesearch, but they're not related to `QueryablePromise`.

**Comment by fergusjordantw** [comment]
Cool, in that case, it's probably fine to not mark it as a breaking change if it's unused across the org, just cleanup at that stage

**Review by fergusjordantw — APPROVED** [review]

**Review by mSobhy90 — APPROVED** [review]
👍

**Pull request merged**